Code/Engine/RendererCore/Components/Implementation/OccluderComponent.cpp

```diff
@@ -0,0 +1,106 @@
+#include <RendererCore/RendererCorePCH.h>
+
+#include <Core/Messages/TransformChangedMessage.h>
+#include <Core/Messages/UpdateLocalBoundsMessage.h>
+#include <Core/WorldSerializer/WorldReader.h>
+#include <Core/WorldSerializer/WorldWriter.h>
+#include <RendererCore/Components/OccluderComponent.h>
+#include <RendererCore/Pipeline/RenderData.h>
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezOccluderComponent, 1, ezComponentMode::Static)
+{
+  EZ_BEGIN_PROPERTIES
+  {
+    EZ_ACCESSOR_PROPERTY("Extents", GetExtents, SetExtents)->AddAttributes(new ezClampValueAttribute(ezVec3(0.0f), {}), new ezDefaultValueAttribute(ezVec3(1.0f))),
+  }
+  EZ_END_PROPERTIES;
+  EZ_BEGIN_MESSAGEHANDLERS
+  {
+    EZ_MESSAGE_HANDLER(ezMsgUpdateLocalBounds, OnUpdateLocalBounds),
+    EZ_MESSAGE_HANDLER(ezMsgExtractOccluderData, OnMsgExtractOccluderData),
+  }
+  EZ_END_MESSAGEHANDLERS;
+  EZ_BEGIN_ATTRIBUTES
+  {
+    new ezCategoryAttribute("Rendering"),
+    new ezBoxVisualizerAttribute("Extents", 1.0f, ezColorScheme::LightUI(ezColorScheme::Blue)),
+    new ezBoxManipulatorAttribute("Extents", 1.0f, true),
+  }
+  EZ_END_ATTRIBUTES;
+}
+EZ_END_COMPONENT_TYPE
+// clang-format on
+
+ezOccluderComponentManager::ezOccluderComponentManager(ezWorld* pWorld)
+  : ezComponentManager<ezOccluderComponent, ezBlockStorageType::FreeList>(pWorld)
+{
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+ezOccluderComponent::ezOccluderComponent() = default;
+ezOccluderComponent::~ezOccluderComponent() = default;
+
+void ezOccluderComponent::SetExtents(const ezVec3& extents)
+{
+  m_vExtents = extents;
+  m_pOccluderObject = nullptr;
+
+  if (IsActiveAndInitialized())
+  {
+    GetOwner()->UpdateLocalBounds();
+  }
+}
+
+void ezOccluderComponent::OnUpdateLocalBounds(ezMsgUpdateLocalBounds& msg)
+{
+  if (GetOwner()->IsStatic())
+    msg.AddBounds(ezBoundingBox(-m_vExtents * 0.5f, m_vExtents * 0.5f), ezDefaultSpatialDataCategories::OcclusionStatic);
+  else
+    msg.AddBounds(ezBoundingBox(-m_vExtents * 0.5f, m_vExtents * 0.5f), ezDefaultSpatialDataCategories::OcclusionDynamic);
+}
+
+void ezOccluderComponent::OnMsgExtractOccluderData(ezMsgExtractOccluderData& msg) const
+{
+  if (IsActiveAndInitialized())
+  {
+    if (m_pOccluderObject == nullptr)
```

Comment:
Since objects are shared now I have to use a mutex during mesh creation, which prevents a race condition. Please have another look. I can change it dispite that, but I don't really see a need for it now.